src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -164,6 +166,53 @@ static ColumnWithTypeAndName readColumnWithStringData(const std::shared_ptr<arro
     return {std::move(internal_column), std::move(internal_type), column_name};
 }
 
+template <typename ArrowArray>
+static ColumnWithTypeAndName readColumnWithJSONData(
+    const std::shared_ptr<arrow::ChunkedArray> & arrow_column,
+    const String & column_name,
+    DataTypePtr type_hint,
+    const FormatSettings & format_settings)
+{
+    static const auto internal_type = type_hint ? type_hint : std::make_shared<DataTypeObject>(DataTypeObject::SchemaFormat::JSON);
+    static const auto serialization = internal_type->getDefaultSerialization();
```

Comment:
The reason of test failures is a data race caused by these fields to be static
```suggestion
   const auto internal_type = type_hint ? type_hint : std::make_shared<DataTypeObject>(DataTypeObject::SchemaFormat::JSON);
   const auto serialization = internal_type->getDefaultSerialization();
```